server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -236,7 +235,12 @@ public ClusterState execute(ClusterState currentState) {
      * Returns the pipeline by the specified id
      */
     public Pipeline getPipeline(String id) {
-        return pipelines.get(id);
+        PipelineHolder holder = pipelines.get(id);
+        if (holder != null) {
+            return holder.pipeline;
+        } else {
+            return null;
```

Comment:
If `holder` var is `null` then this method should also return `null`. This behaviour is the same as before this change. If this null check doesn't happen then that would cause an npe in the case there is no pipeline for an id.